crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -223,6 +223,35 @@ impl Format<PyFormatContext<'_>> for MaybeParenthesizeExpression<'_> {
                     }
                 }
             },
+            OptionalParentheses::IfFits => match parenthesize {
```

Comment:
I'm not very happy with this name. The branches are the same as for `Never` except for `IfBreaks`.